src/test/unit/requests/hearingRequests.test.ts

```diff
@@ -33,3 +42,62 @@ describe('Hearing get requests', () => {
     expect(await hearingRequests.getHearingsByCaseName('bob')).toStrictEqual([]);
   });
 });
+
+describe(`getHearingByCaseReferenceNumber(${validCaseNo})`, () => {
+  beforeEach(() => {
+    stub.withArgs('/hearings/case-number/').resolves(Promise.reject(errorResponse));
+    stub.withArgs('/hearings/case-number/fail').resolves(Promise.reject(errorRequest));
+    stub.withArgs('/hearings/case-number/ABC12345').resolves({data: subscriptionsCaseData});
+  });
+
+  it('should return list of cases', async () => {
+    return hearingRequests.getHearingByCaseReferenceNumber(validCaseNo).then(data => {
+      expect(data).toBe(subscriptionsCaseData);
+    });
+  });
+
+  it('should return list of 1 case', () => {
+    return hearingRequests.getHearingByCaseReferenceNumber(validCaseNo).then(data => {
+      expect((data? 1:0)).toBe(1);
+    });
+  });
+
+  it('should have mocked object in the cases list', () => {
+    return hearingRequests.getHearingByCaseReferenceNumber(validCaseNo).then(data => {
+      expect((data === subscriptionsCaseData) ? 1 : 0).toBe(1);
+    });
+  });
+
+  it('should return empty list if request fails', async () => {
+    expect(await hearingRequests.getHearingByCaseReferenceNumber('')).toStrictEqual(null);
+  });
+
+  it('should return empty list if request fails', async () => {
+    expect(await hearingRequests.getHearingByCaseReferenceNumber('fail')).toStrictEqual(null);
+  });
+});
+
+describe('non existing subscriptions getHearingByCaseReferenceNumber error request', () => {
+  stub.withArgs('/hearings/case-number/12345').resolves(Promise.reject(errorRequest));
+  it(`should have only cases for case reference ${validCaseNo}`, () => {
+    return hearingRequests.getHearingByCaseReferenceNumber(validCaseNo).then(data => {
+      expect((data[0].caseNumber === validCaseNo ? 1 : 0)).toBe(1);
```

Comment:
why ternary operator here? I believe it can be expect(data[0].caseNumber).toEqual(validCaseNo)

Reply:
fixed.